sdks/python/apache_beam/io/filebasedsource.py

```diff
@@ -365,9 +365,13 @@ def process(self, element, *args, **kwargs):
     source = self._source_from_file(metadata.path)
     # Following split() operation has to be performed to create a proper
     # _SingleFileSource. Otherwise what we have is a ConcatSource that contains
-    # a single _SingleFileSource. ConcatSource.read() expects a RangeTraker for
+    # a single _SingleFileSource. ConcatSource.read() expects a RangeTracker for
     # sub-source range and reads full sub-sources (not byte ranges).
-    source = list(source.split(float('inf')))[0].source
+    source_list = list(source.split(float('inf')))
+    # Handle the case of an empty source.
+    if not source_list:
+      return
```

Comment:
Hmm, can you explain when this will occur in practice and create a JIRA ?

split() method is expected to return a set of sources that read the same set of data (combined) as the original source.